Move react menus to layout folder

antonioortizpola opened the pull request:
Move react 'menus' from header to layout folder to use them as more general components in a left panel or a site map

Fix #8142

-   Please make sure the below checklist is followed for Pull Requests.
The changes are just some orders, all test should pass.
-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

<!--
Please also reference the issue number in a commit message to [automatically close the related Github issue](https://help.github.com/articles/closing-issues-via-commit-messages/)

Note: It is also possible to add `[skip ci]` to your commit message to skip Travis tests
-->

antonioortizpola commented:
I know, even for my project this means some changes. If you think of a better way to handle this I am all ears, if you think it is too much o a change for a little benefit I understand, this is a little closer to the way i see the angular code, but i am new in react and in JHipster, so I trust in your opinions.

jdubois commented:
I totally agree with both of you (this looks good, then it's going to break apps). But also, I know how it works with PR that are opened for too long: they are impossible to merge. So waiting for JHipster 6 looks very risky to me, as we have nothing planned yet.

DanielFran commented:
@antonioortizpola Can you please fix the conflicts?

antonioortizpola commented:
@DanielFran, sure, just give me one or two days since i have some work to do, but i will be happy to find some time in the night to help

antonioortizpola commented:
Sorry for taking so long to fix the conflicts, i had tons of work and i could not make time for this until today.

I had some difficulties with the rebase and merge, but i thing i successfully cleaned up my mess, i ran the tests `npm test` and `gradlew test` and everything seemed ok, i am waiting for the travis test to run, if everything it is ok, i will remove WIP from the title to let you know the merge is ready

avdev4j commented:
Hi @antonioortizpola 
I think you failed a part of your rebase especially the generator-base :D 
I think it's not easy to do, good luck. Don't hesitate to ping if you need help for the review.

antonioortizpola commented:
@avdev4j OMG the pace of the changes are fast, yesterday i fixed all the conflicts, the branch was ready, but the last commits of @jdubois generate just one in generator-base.js.

I fixed the new conflicts, thankfully this time it was really easy, i hope this time it holds up for revision.

The Travis tests are still running, but if you could help me reviewing the changes to check if everything looks ok and if it is worth to merge my changes.

antonioortizpola commented:
2 test failed, but i do not think is something to do with my changes

![image](https://user-images.githubusercontent.com/11655127/53270132-79f8c000-36b0-11e9-9db0-bbd50c06c23c.png)

In the logs it looks like is something to do with the database

> The last packet sent successfully to the server was 0 milliseconds ago. The driver has not received any packets from the server

Should i make a dumb change to trigger the tests again?, the test before this one passed, but I realized i had some years in the licence file as 2018, so i updated them, this was the only change, so i am confident that the errors are not for my changes

ruddell commented:
I restarted the failing jobs

antonioortizpola commented:
Everything seems fine now! :D I will wait for your observations

avdev4j commented:
For me it's ok about generator-base I guess you don't need to change it and the last modifications were made by error (rebasing an old PR is not very easy and funny I admit).

Great Job 👍

jdubois commented:
@deepu105 I understand that you agreed about this change, is it correct? If true, I could merge this in the next couple of days

DanielFran commented:
@antonioortizpola Just 1 minor conflit to fix before @jdubois can merge it.

antonioortizpola commented:
@DanielFran Sure, I can fix it tonight, i was waiting for confirmation since the changes in the repo are fast and i did not want to be fixing conflicts again and again.

Thanks a lot, I will let you know when it is ready

antonioortizpola commented:
@DanielFran the conflict is fixed :D, but I do not see the tests running

DanielFran commented:
@deepu105 Can you please validate if this can be merged, please?

DanielFran commented:
Closing/Reopening to launch tests

DanielFran commented:
@antonioortizpola there is some issues, can you please fix needle-client tests?
See https://travis-ci.org/jhipster/generator-jhipster/jobs/504269836

antonioortizpola commented:
I do not think this has something to do with my changes, the tests was passed when i fixed the issues, checking the error, i do not know how my change could cause that test to fail
![image](https://user-images.githubusercontent.com/11655127/54145505-c7389980-43f3-11e9-9a16-297ed721aa28.png)
In fact, the react test passed.

DanielFran commented:
Closing/Reopening to launch tests

antonioortizpola commented:
@DanielFran Ok, so sorry, i did not catch that one

DanielFran commented:
@jdubois @pascalgrimaud Any clue? It disapears also to me since few days ago...

pascalgrimaud commented:
it's the migration from travis-ci.org to travis-ci.com -> some autorizations are in pending to be accepted by the organization

antonioortizpola commented:
Ok, all test passed, however the change was a little odd

![image](https://user-images.githubusercontent.com/11655127/54662023-0fb22000-4aa2-11e9-88c4-d74bf6c9b3d3.png)

Apparently JsLisnt did not liked my multiline string, changing to a one liner made the problem dissapear
![image](https://user-images.githubusercontent.com/11655127/54662208-e2b23d00-4aa2-11e9-93ea-676726a9c1c5.png)

DanielFran commented:
Closing/Reopenning

DanielFran commented:
Thanks @antonioortizpola and sorry for this long time waiting...

antonioortizpola commented:
It is ok, i understand how hard can be keep a project this magnitude and why you need to review all the details, I still realy hope to help more in the future!
